src/classes/SObjectRepositoryMocks.cls

```diff
@@ -0,0 +1,114 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class SObjectRepositoryMocks {
+    public virtual class Base extends CrudMock implements ISObjectRepository {
+        protected List<SObject> records;
+        protected List<Id> recordIdList;
+        protected List<QueryPredicate> predicates;
+
+        public Base() {
+            this.records = new List<SObject>();
+        }
+
+        public Base(List<SObject> records) {
+            this.records = records;
+        }
+        public Base with (List<SObject> records) {
+            this.records = records;
+            return this;
+        }
+
+        //SOQL
+        public virtual SObject getById(Id recordId) {
+            return this.getById(new List<Id>{recordId})[0];
+        }
+
+        public virtual List<SObject> getById(List<Id> recordIdList) {
+            this.recordIdList = recordIdList;
+
+            if(this.records == null) {
```

Comment:
Can we smoosh this down to one line?
`if(this.records == null) this.createMockObjects();`